scripts/include/helpers.inc.php

```diff
@@ -121,45 +121,98 @@ function mergeXliffDomToAtom(DOMDocument $domA, DOMDocument $domB): DOMDocument
 
 function mergeXliffDomToWeblate(DOMDocument $domA, DOMDocument $domB): DOMDocument
 {
-  // Elements in A but not in B -> copy to B
-  $sourcesA = $domA->getElementsByTagName('source');
-  $sourcesB = $domB->getElementsByTagName('source');
+  foreach ($domB->getElementsByTagName('trans-unit') as $element) {
+    if ($element->hasAttribute('id')) {
+        $element->setIdAttribute('id', true);
+    }
+  }
+
+  // Elements in A AND B -> check source string and if different -> replace in B
+  $transUnitsA = $domA->getElementsByTagName('trans-unit');
+  $transUnitsB = $domB->getElementsByTagName('trans-unit');
```

Comment:
Suggestion: If you move the foreach above (lines 124-128) after this assignment and loop through `$transUnitsB` instead it might save some resources by avoiding searching the DOM twice.

Reply:
Good idea. Refactored to collect this into one loop.